### PR TITLE
A few updates to HybridConnectionListener.closeAsync

### PR DESCRIPTION
**Pull request by dlstucki**
- Make a few updates to HybridConnectionListener.closeAsync.
- Make the appropriate API methods public on InputQueue (class is still package).

### CODE REVIEW

File: src/main/java/com/microsoft/azure/relay/HybridConnectionListener.java [hunk 1]
```diff
@@ -276,46 +276,42 @@ public HybridConnectionListener(String connectionString, String path) throws URI
...
-		}, EXECUTOR)
-		.thenCompose(closeTasks -> {
-			return CompletableFuture.allOf((CompletableFuture<Void>[]) closeTasks).thenRun(() -> {
+		}
+		
+		return CompletableFuture.allOf(closeTasks)
+			.thenCompose(voidResult -> this.controlConnection.closeAsync(timeoutHelper.remainingTime()))
+			.whenComplete((voidResult, ex) -> {
 				this.isOnline = false;
 				RelayLogger.logEvent("closed", this);
```
Comment (bainian12345) [inline src/main/java/com/microsoft/azure/relay/HybridConnectionListener.java]: >RelayLogger.logEvent("closed", this); [](start = 4, length = 37)

maybe this log should be at the very end?

Comment (dlstucki) [reply src/main/java/com/microsoft/azure/relay/HybridConnectionListener.java]: I can move it after connectionInputQueue.dispose(), or is there somewhere else you'd like it to be?

---
In reply to: [258612564](https://github.com/Azure/azure-relay-java/pull/15#discussion_r258612564) [](ancestors = 258612564)

Comment (bainian12345) [reply src/main/java/com/microsoft/azure/relay/HybridConnectionListener.java]: That works

---
In reply to: [258613118](https://github.com/Azure/azure-relay-java/pull/15#discussion_r258613118) [](ancestors = 258613118,258612564)

Comment (dlstucki) [reply src/main/java/com/microsoft/azure/relay/HybridConnectionListener.java]: Fixed in Iteration 2.

---
In reply to: [258613625](https://github.com/Azure/azure-relay-java/pull/15#discussion_r258613625) [](ancestors = 258613625,258613118,258612564)

**Pull request merged**